linux/hci/hci.go

```diff
@@ -411,6 +425,24 @@ func (h *HCI) sktLoop() {
 				_ = logger.Error("skt: %v", err)
 			} else {
 				log.Printf("skt: %v", err)
+				// close current hci socket
+				err = h.Close()
+				if err != nil {
+					return
+				}
+
+				// create hci socket again
+				time.Sleep(3 * time.Second)
+				skt, err := socket.NewSocket(h.id)
+				if err != nil {
+					return
+				}
+				h.skt = skt
+
+				// notify canceling to send hci command
+				close(h.cancel)
+				h.cancel = make(chan bool)
+				h.err = ErrHCIHandlePacket
```

Comment:
Due to the infrequency of reproduction, the sktLoop was rewritten as follows to confirm that it works as expected.

`sudo go run examples/basic/extended_scanner/main.go`

```go

var count = 0

func (h *HCI) sktLoop() {
	b := make([]byte, 4096)
	defer close(h.done)
	for {
		n, err := h.skt.Read(b)
		if n == 0 || err != nil {
			if err == io.EOF {
				h.err = err //callers depend on detecting io.EOF, don't wrap it.
			} else {
				h.err = fmt.Errorf("skt: %s", err)
			}
			return
		}
		p := make([]byte, n)
		copy(p, b)

		if err := h.handlePkt(p); err != nil {
			// Some bluetooth devices may append vendor specific packets at the last,
			// in this case, simply ignore them.
			if strings.HasPrefix(err.Error(), "unsupported vendor packet:") {
				_ = logger.Error("skt: %v", err)
			} else {
				log.Printf("skt: %v", err)
				// close current hci socket
				err = h.Close()
				if err != nil {
					return
				}

				// create hci socket again
				time.Sleep(3 * time.Second)
				skt, err := socket.NewSocket(h.id)
				if err != nil {
					return
				}
				h.skt = skt

				// notify canceling to send hci command
				h.err = ErrHCIHandlePacket
				close(h.cancel)
				h.cancel = make(chan bool)
				continue
			}
		}

		count = count + 1

		if count%10 == 0 {
			if count < 11 {
				// close current hci socket
				err = h.Close()
				if err != nil {
					return
				}
				// create hci socket again
				time.Sleep(3 * time.Second)
				skt, err := socket.NewSocket(h.id)
				if err != nil {
					return
				}
				h.skt = skt

				// notify canceling to send hci command
				h.err = ErrHCIHandlePacket
				close(h.cancel)
				h.cancel = make(chan bool)
				continue
			}
		}
	}
}
```